src/client.rs

```diff
@@ -41,14 +44,23 @@ impl FederationClient {
         let connector = MatrixConnector::with_default_resolver()?;
 
         Ok(FederationClient {
-            client: Client::builder().build(connector),
+            client: Client::builder(TokioExecutor::new()).build(connector),
         })
     }
 
-    pub async fn request(&self, mut req: Request<Body>) -> Result<Response<Body>, Error> {
+    pub async fn request(
+        &self,
+        mut req: Request<Full<Bytes>>,
+    ) -> Result<Response<Full<Bytes>>, Error> {
         req = handle_delegated_server(&self.client, req).await?;
 
-        Ok(self.client.request(req).await?)
+        match self.client.request(req).await {
+            Ok(resp) => match BodyExt::collect(resp.into_body()).await {
+                Ok(collected_bytes) => Ok(Response::new(Full::new(collected_bytes.to_bytes()))),
+                Err(err) => Err(err.into()),
+            },
+            Err(err) => Err(err.into()),
+        }
```

Comment:
We have to convert from `http::Response<hyper::body::Incoming>` to `http::Response<Full<Bytes>>` in a few places

Reply:
Updated to use `http::Response<hyper::body::Incoming>` directly :fast_forward: